00X-OFEP-provider-flag-metadata.md

```diff
@@ -0,0 +1,37 @@
+## Surfacing flag metadata
+
+## State: DRAFTING
+This proposal lays out a mechanism for flag providers to surface arbitrary flag metadata to Open Feature, and for hooks to access this metadata.
+
+
+## Background
+
+Flag providers maintain metadata about a feature flag, and there are scenarios where a hook might benefit from being able to access this metadata. For example, a flag provider could surface a `management-url` attribute which an open-telemetry hook could then add to an otel span, allowing someone viewing a trace for a feature-flagged operation to easily navigate to a webpage describing that flag in detail.
```

Comment:
> This use case would appear to me as more of a global than per flag use case. Where you join the `management-url` of the `global` config with the specific flag key to add to otel span.
> 
> Are there additional use case examples that can be added for individual metadata on a flag alone?

I would have though the management url would be per-flag: `https://app.flagrsus.com/flags/1412351` or whatever. I would expect a provider would want it to be an opaque url rather than having each hook build the URL itself.

Other use cases off the top of my head:
- owner: which team or person owns this flag, so I can go ping them and ask whether their flag could be related to some operational issue
- expiration date: should this flag have been retired by now? I've seen teams log warning or even crash hard if their code is referencing stale flags. "Feature flag debt" is a very real thing.
- subsystem: which subsystem is supposed to be controlled by this flag, so I can enforce rules regarding which flags are used where. For example, I might want to enforce that client-side code isn't using a server-side flag, or web-app code isn't using a mobile flag, or whatever.

Reply:
Another example: https://github.com/open-feature/ofep/issues/34

> As it relates to flag lifecycle & experimentation, they have a pattern where sometimes they'll query for a flag value early and actually use it much later. In that time, the values may have swapped. As such, I think they want the ability to include other metadata along with the EvaluationDetails (like timestamp of fetch, etc)